service/history/queues/executable_factory.go

```diff
@@ -77,6 +78,7 @@ func NewExecutableFactory(
 	dlqEnabled dynamicconfig.BoolPropertyFn,
 	attemptsBeforeSendingToDlq dynamicconfig.IntPropertyFn,
 	dlqInternalErrors dynamicconfig.BoolPropertyFn,
+	dlqErrorSubStrings dynamicconfig.StringPropertyFn,
```

Comment:
can we reuse ExecutableParams here?

Reply:
we are using ExecutableParams here. But this property is also a member of Executable. This is in line with other properties that are passed to this object.

Now that we are passing 4 different config values to Executable. It makes sense to just pass a pointer to Config object. For that, I will have to refactor all the tests. I can do that in a different PR if that makes sense.